### PR TITLE
Psr0Fixer - fix bug for fixing file with long extension like .class.php

### PR DESCRIPTION
**Pull request by keradus**
Reported [here](https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/995#issuecomment-83944377)

### CODE REVIEW

**Comment by keradus** [comment]
ping @dzgrief

**Comment by GrahamCampbell** [comment]
We shouldn't allow .class.php. That is a violation of psr0...

**Comment by ceeram** [comment]
How does such a filename even get loaded with psr0?

2015-03-21 13:34 GMT+01:00 Dariusz Rumiński <notifications@github.com>:

**Comment by GrahamCampbell** [comment]
> How does such a filename even get loaded with psr0?

Exactly, it doesn't.

I'm massively :-1: on this. We can't brand something that's not psr0 under psr0...

**Comment by keradus** [comment]
Yes, I hate `.class.php` too.


But have a deeper look. For now we have that file `Foo.class.php`, which contains:
`<?php class Foo {}`.

And our current stable release changes the code into:
`<?php class Foo.class {}`.
which is illegal, and either crash (when using linter), either create illegal code (without linter)

**Comment by ceeram** [comment]
Then the fixer is handling it incorrect. Still .class.php filenames are not
psr0 autoloading compliant.

2015-03-21 13:50 GMT+01:00 Dariusz Rumiński <notifications@github.com>:

**Comment by keradus** [comment]
Any constructive criticism?
Leaving the crash alone is not an option.

**Comment by ceeram** [comment]
Only constructive criticism allowed in this project? I was just mentioning
something that seemed to go by unnoticed, which imo was too important to
not criticize.


2015-03-21 13:53 GMT+01:00 Dariusz Rumiński <notifications@github.com>:

**Comment by GrahamCampbell** [comment]
@keradus, @ceeram and I are basically saying that your fix is incorrect. You are fixing the code to violate psr0.

**Comment by keradus** [comment]
Hey guys.
No, not only constructive criticism is allowed.
But indeed, there is no point to say "it is a bad solution" over and over again, the first one was enought.
Instead of saying same thing few more time it would be cool to also bring something new into subject.

**Comment by keradus** [comment]
Rewritten to change `supports` method. Now file like `Foo.class.php` will be ignored by that fixer.

**Comment by GrahamCampbell** [comment]
@keradus That implementation is still wrong. We are calling this a psr0 fixer. This is not fixing to psr0, so it's totally wrong.

**Comment by GrahamCampbell** [comment]
Our psr0 fixer is wrong anyway imo.

We should be renaming files to match the class names, and should be renaming `Foo.class.php` to `Foo.php`.

File: Symfony/CS/Fixer/PSR0/Psr0Fixer.php [hunk 1]
```diff
@@ -144,7 +145,9 @@ public function getDescription()
      */
     public function supports(\SplFileInfo $file)
     {
-        if ('php' !== pathinfo($file->getFilename(), PATHINFO_EXTENSION)) {
+        $filenameParts = explode('.', $file->getBasename(), 2);
+
+        if ('php' !== $filenameParts[1]) {
```
Comment (gharlan) [inline Symfony/CS/Fixer/PSR0/Psr0Fixer.php]: When the file does not have any extension, `$filenameParts[1]` will not be set.

Comment (keradus) [inline Symfony/CS/Fixer/PSR0/Psr0Fixer.php]: good point.

**Comment by keradus** [comment]
> That implementation is still wrong. We are calling this a psr0 fixer. This is not fixing to psr0, so it's totally wrong.

Then don't know about what you are talking.
Current Psr0Fixer cashes on `Foo.class.php` file. It already ignore non .php file, but does not ignore .class.php files. Now it will does and stop crushing. Nothing more.
If you have any thoughts about incorrect way of fixing files itself then this is an different issue and please send separate PR.


> We should be renaming files to match the class names, and should be renaming Foo.class.php to Foo.php.

Nope, we do not rename files.

**Comment by GrahamCampbell** [comment]
> Nope, we do not rename files.

I know, but I think we should be. That's what I'm saying. Could we rewrite this fixer in 2.0?

**Comment by keradus** [comment]
It is another problem that need to be discussed separately.
Probably with setting risky for it.

**Comment by GrahamCampbell** [comment]
The current fixer is more risky though. It's changing the code.

**Comment by keradus** [comment]
Just like I said...

**Comment by ceeram** [comment]
This issue exactly points out why filenames should be adapted to match the classnames. PSR0 autloaders also work with classnames and then know what the filename must be to match the classname.

**Comment by GrahamCampbell** [comment]
@ceeram I agree.

**Comment by gharlan** [comment]
> It is another problem that need to be discussed separately.

-> #117

**Comment by dzgrief** [comment]
Problem solved, thank you.

In the actual development, we may use some open source framework, the framework defines its own naming rules, although these rules do not conform to the psr0, but if you want to use these frameworks, can only obey the rules of the framework of naming, rather than psr0.

**Comment by dzgrief** [comment]
##### Now have found another problem: the code comments [@return] will word wrap.

```
/**
 * encrypt string
 * @param  string $key        [description]
 * @param  string $distracter [description]
 * @return string             [description]
 */
public function encrypt($key, $distracter = null)
{
}
```

##### Will convert

```
/**
 * encrypt string.
 *
 * @param string $key        [description]
 * @param string $distracter [description]
 *
 * @return string [description]
 */
public function encrypt($key, $distracter = null)
{
}
```

**Comment by keradus** [comment]
Please keep the thread clean and report each issue as separate thread to discuss it separately and one at time.

**Pull request closed without merging**